package/kedro_viz/autoreload_file_filter.py

```diff
@@ -0,0 +1,88 @@
+"""
+This module provides a custom file filter for autoreloading that filters out files based on allowed
+file extensions and patterns specified in a .gitignore file.
+"""
+
+import logging
+from pathlib import Path
+from typing import Optional, Set
+
+from pathspec import PathSpec
+from watchfiles import Change, DefaultFilter
+
+logger = logging.getLogger(__name__)
+
+
+class AutoreloadFileFilter(DefaultFilter):
+    """
+    Custom file filter for autoreloading that extends DefaultFilter.
+    Filters out files based on allowed file extensions and patterns specified in a .gitignore file.
+    """
+
+    allowed_extensions: Set[str] = {".py", ".yml", ".yaml", ".json"}
+
+    def __init__(self, base_path: Optional[Path] = None):
+        """
+        Initialize the AutoreloadFileFilter.
+
+        Args:
+            base_path (Optional[Path]): The base path to set as the current working directory
+                for the filter.
+        """
+        self.cwd = base_path or Path.cwd()
+
+        # Call the superclass constructor
+        super().__init__()
+
+        # Load .gitignore patterns
+        gitignore_path = self.cwd / ".gitignore"
+        try:
+            with open(gitignore_path, "r", encoding="utf-8") as gitignore_file:
+                ignore_patterns = gitignore_file.read().splitlines()
```

Comment:
so this would get all files that are not part of git ? 
This should be good but do we need any check for hidden files as well ?

Reply:
> so this would get all files that are not part of git ?

Yes

> do we need any check for hidden files as well ?

This is already handle by `DefaultFilter` class internally for hidden files or file start with `.`